### ISSUE TITLE
Add support for more interval types for delta aggregation

### ISSUE THREAD

**Issue by nikmagini** [dmwm/PhedexReplicaMonitoring#11]
Currently, the 'delta' aggregation function works only with a fixed number of days as argument of the --interval option. e.g. --interval 31

For long aggregations, it would be convenient to add support for different type of time intervals, for example

--interval 1months

So that we don't need to take into account if the month has 28/29/30/31 days when we define the interval for monthly aggregations

**Comment by vkuznet** [comment]
Patch has been provided in https://github.com/dmwm/PhedexReplicaMonitoring/pull/13
Now user can specify --interval 1m, 1d, 1w, etc. I allowed to use w, week, weeks, d, day, days, m, month, months as suffixes to a given number. The specified human based interval is converted to integer in form of days.

**Comment by vkuznet** [comment]
Nicolo, I want to be sure I implemented correctly your request. So, if someone will provide --interval 2m
(means 2months) I take current date and subtract 2 months from it. Internally I do the calculation of number of days in that interval, e.g. 60 (sometimes 61, etc.) and use it as number of days in interval option. Does this logic is correct?

**Comment by nikmagini** [comment]
I was thinking of something slightly different: taking the first day of the month for each selected month.
To be able e.g. to run on an entire year at once, getting statistics for each 1st day of the month (I admit that this will cause a variable interval length...)

**Comment by vkuznet** [comment]
How about we use --fromdate to specify the initial date and --interval Nmonth will ensure that month are added to fromdate value. Doing this way you can do entire year as following
--fromdate=2016-01-01 --todate=2017-01-01 --interval=1m
will provide ranges for every 1st day of the months in 2016.

**Comment by nikmagini** [comment]
Sounds good

**Comment by vkuznet** [comment]
Nicolo, here is what new code is produce, please confirm that this is correct.

for 1m interval between specified dates:
```
('fromdate', '2016-01-01', 'todate', '2016-03-30', 'interval', '1m')
('boundict', {1: ['2016-01-01', '2016-01-31'], 2: ['2016-02-01', '2016-03-01']})
```

for numeric interval between specified dates:
```
('fromdate', '2016-01-01', 'todate', '2016-03-30', 'interval', 31)
('boundict', {1: ['2016-01-01', '2016-01-31'], 2: ['2016-02-01', '2016-03-03']})
```

I printed boundict in a form as current code uses, i.e. it is a dict of indexes with time-ranges. Please note that last pair may not be aligned with todate value, because todate is just a boundary for loop rather last value in boundict.

**Comment by nikmagini** [comment]
Hi Valentin,

for the first case, I think I expect this instead:

'''('fromdate', '2016-01-01', 'todate', '2016-03-30', 'interval', '1m')
('boundict', {1: ['2016-01-01', '2016-01-31'], 2: ['2016-02-01', '2016-02-29'], 3:  ['2016-03-01', '2016-03-30']})'''

And for the second case, the second interval in your example is 32 days instead of 31. So I would expect this instead:

'''('fromdate', '2016-01-01', 'todate', '2016-03-30', 'interval', 31)
('boundict', {1: ['2016-01-01', '2016-01-31'], 2: ['2016-02-01', '2016-03-02']})'''

(also note issue #15 where I propose to use the start instead of the end of each time-range for aggregation)

**Comment by vkuznet** [comment]
Ok, here is revisited code output:
```
('fromdate', '2016-01-01', 'todate', '2016-12-31', 'interval', '1m')
('boundic', {1: ['2016-01-01', '2016-01-31'], 2: ['2016-02-01', '2016-02-29'], 3: ['2016-03-01', '2016-03-31'], 4: ['2016-04-01', '2016-04-30'], 5: ['2016-05-01', '2016-05-31'], 6: ['2016-06-01', '2016-06-30'], 7: ['2016-07-01', '2016-07-31'], 8: ['2016-08-01', '2016-08-31'], 9: ['2016-09-01', '2016-09-30'], 10: ['2016-10-01', '2016-10-31'], 11: ['2016-11-01', '2016-11-30']})
```

```
('fromdate', '2016-01-01', 'todate', '2016-12-31', 'interval', 31)
('boundic', {1: ['2016-01-01', '2016-01-31'], 2: ['2016-02-01', '2016-03-02'], 3: ['2016-03-03', '2016-04-02'], 4: ['2016-04-03', '2016-05-03'], 5: ['2016-05-04', '2016-06-03'], 6: ['2016-06-04', '2016-07-04'], 7: ['2016-07-05', '2016-08-04'], 8: ['2016-08-05', '2016-09-04'], 9: ['2016-09-05', '2016-10-05'], 10: ['2016-10-06', '2016-11-05'], 11: ['2016-11-06', '2016-12-06']})
```

**Comment by vkuznet** [comment]
Here is full output for d,w,m and integers:
```
('fromdate', '2016-01-01', 'todate', '2016-12-31', 'interval', '1m')
('boundic', {1: ['2016-01-01', '2016-01-31'], 2: ['2016-02-01', '2016-02-29'], 3: ['2016-03-01', '2016-03-31'], 4: ['2016-04-01', '2016-04-30'], 5: ['2016-05-01', '2016-05-31'], 6: ['2016-06-01', '2016-06-30'], 7: ['2016-07-01', '2016-07-31'], 8: ['2016-08-01', '2016-08-31'], 9: ['2016-09-01', '2016-09-30'], 10: ['2016-10-01', '2016-10-31'], 11: ['2016-11-01', '2016-11-30']})
('fromdate', '2016-01-01', 'todate', '2016-12-31', 'interval', 31)
('boundic', {1: ['2016-01-01', '2016-01-31'], 2: ['2016-02-01', '2016-03-02'], 3: ['2016-03-03', '2016-04-02'], 4: ['2016-04-03', '2016-05-03'], 5: ['2016-05-04', '2016-06-03'], 6: ['2016-06-04', '2016-07-04'], 7: ['2016-07-05', '2016-08-04'], 8: ['2016-08-05', '2016-09-04'], 9: ['2016-09-05', '2016-10-05'], 10: ['2016-10-06', '2016-11-05'], 11: ['2016-11-06', '2016-12-06']})
('fromdate', '2016-01-01', 'todate', '2016-12-31', 'interval', '4w')
('boundic', {1: ['2016-01-01', '2016-01-28'], 2: ['2016-01-29', '2016-02-25'], 3: ['2016-02-26', '2016-03-24'], 4: ['2016-03-25', '2016-04-21'], 5: ['2016-04-22', '2016-05-19'], 6: ['2016-05-20', '2016-06-16'], 7: ['2016-06-17', '2016-07-14'], 8: ['2016-07-15', '2016-08-11'], 9: ['2016-08-12', '2016-09-08'], 10: ['2016-09-09', '2016-10-06'], 11: ['2016-10-07', '2016-11-03'], 12: ['2016-11-04', '2016-12-01'], 13: ['2016-12-02', '2016-12-29']})
('fromdate', '2016-01-01', 'todate', '2016-12-31', 'interval', '30d')
('boundic', {1: ['2016-01-01', '2016-01-30'], 2: ['2016-01-31', '2016-02-29'], 3: ['2016-03-01', '2016-03-30'], 4: ['2016-03-31', '2016-04-29'], 5: ['2016-04-30', '2016-05-29'], 6: ['2016-05-30', '2016-06-28'], 7: ['2016-06-29', '2016-07-28'], 8: ['2016-07-29', '2016-08-27'], 9: ['2016-08-28', '2016-09-26'], 10: ['2016-09-27', '2016-10-26'], 11: ['2016-10-27', '2016-11-25'], 12: ['2016-11-26', '2016-12-25']})
```

**Comment by vkuznet** [comment]
this output consists with choice of
```
[fromdate, fromdate+N], [fromdate+N+1, fromdate+2N], [fromdate+2N+1, fromdate+3N]
```
from #15

**Comment by vkuznet** [comment]
I implemented discussed interval range, see PR #16. Closing the ticket.
